src/index.ts

```diff
@@ -69,6 +69,22 @@ export function getFormattedSrcLink(filepath: string, repoUrl?: RepoUrl) {
   return srcLink
 }
 
+export function prepareTodosForDanger(keywords: string[] | undefined, addedText: string, removedText: string, filepath: string, repoUrl: string | ((filepath: string) => string) | undefined | boolean, keywordMatches: PlainObject<string[]>) {
+  if (keywords === undefined) return
+  keywords.forEach(keyword => {
+    const addedMatches = getMatches(addedText, keyword)
+    const removedMatches = getMatches(removedText, keyword)
+    const srcLink = getFormattedSrcLink(filepath, repoUrl)
+
+    addedMatches.forEach(match => {
+      keywordMatches[keyword].push(`\`\`${match}\`\`: ${srcLink}`)
```

Comment:
If we are going to move this to a function, we could return a new   matches object instead of mutating a parameter?

Reply:
> If we are going to move this to a function, we could return a new matches object instead of mutating a parameter?

I think I might have been wrong with this suggestion :sweat: I forgot `keywordMatches` was being used for the whole Promise.All loop :man_facepalming: Sorry.

You can revert the commit or keep it if you think that is better. Might be an unnecessary lodash.merge

Reply:
I think this is fine to let it as is. There will be just few TODO's, so merge overhead should be worth for cleaner code.